### PR TITLE
Configure Github Actions to Build and Run Tests

### PR DESCRIPTION
**Pull request by jurvis**
## Purpose
Set up workflows for CI with Github Actions. Addresses #63.

This is dependent on #69 which brings most of our build toolchain to be runnable directly from the Xcode project. 

## Implementation
Create a `build-frameworks` workflow which first builds the files and then runs tests.

You can see an example of the results here: https://github.com/jurvis/libwally-swift/actions/runs/2252283407

## Demo
image

## Next Steps
Write a workflow on release tag that will build an `xcframework` and upload a ZIP to GitHub.

## Additional Notes
I wasn't able to get the CI build with iOS Device working (see: https://github.com/jurvis/libwally-swift/runs/6244574594?check_suite_focus=true), but I was able to confirm on my computer that the framework builds okay for device as target. 

I think there may be a separate issue with iPhoneSDK15.2, but not sure. Attaching gist of `config.log` error here: https://gist.github.com/jurvis/f45901634b5bab5b73db5ad6edf125cd (see line 984 to 1003). Will like some ideas if anyone has any.

### CODE REVIEW

**Comment by Sjors** [comment]
Nice! I'll review #69 first.

File: build-libwally.sh [hunk 1]
```diff
@@ -50,6 +50,11 @@ if [[ ${ACTION:-build} = "build" || $ACTION = "install" ]]; then
     export CFLAGS="-O3 ${ARCHES[@]} -fembed-bitcode -m${TARGET_OS}-version-min=11.0 -isysroot `xcrun -sdk ${PLATFORM_NAME} --show-sdk-path`"
     export CXXFLAGS="-O3 ${ARCHES[@]} -fembed-bitcode -m${TARGET_OS}-version-min=11.0 -isysroot `xcrun -sdk ${PLATFORM_NAME} --show-sdk-path`"
 
+    # CPPFLAGS only required for 64-bit host machines, doing this on arm64 machines will cause a duplicate symbols error.
```
Comment (Sjors) [inline build-libwally.sh:54]: arm64 is 64 bit too :-)

Did you mean "x86 host machines"?

Comment (jurvis) [reply build-libwally.sh:53]: oh 🤦  good point

**Comment by jurvis** [comment]
Latest Github Action build with iOS Device + x86_64 host: https://github.com/jurvis/libwally-swift/actions/runs/2254436441

**Comment by Sjors** [comment]
95c91e7d0bfce894e49a2cb652d201752b8be872 looks good

**Comment by jurvis** [comment]
oops i closed this by accident

**Comment by Sjors** [comment]
It didn't let you reopen it?

**Comment by jurvis** [comment]
nope :( the "Reopen" button is greyed out for me

**Pull request closed without merging**